stackrox-container-image-scanner/src/main/java/com/stackrox/jenkins/plugins/services/ApiClientFactory.java

```diff
@@ -156,4 +163,40 @@ private static SSLContext getSslContext(TrustManager[] trustManagers) throws NoS
         sslContext.init(null, trustManagers, null);
         return sslContext;
     }
+
+    public static class UserAgentInterceptor implements Interceptor {
+        public static final String STACKROX_CONTAINER_IMAGE_SCANNER = "stackrox-container-image-scanner";
+
+        @NotNull
+        @Override
+        public Response intercept(@NotNull Chain chain) throws IOException {
+            String value = String.format("%s/%s (%s; %s) %s",
+                    STACKROX_CONTAINER_IMAGE_SCANNER,
+                    getVersion(),
+                    System.getProperty("os.name"),
+                    System.getProperty("os.arch"),
+                    "CI"
+            );
+
+            return chain.proceed(
+                    chain.request()
+                            .newBuilder()
+                            .header("User-Agent", value)
+                            .build()
+            );
+        }
+
+        String getVersion() {
+            Jenkins jenkins = Jenkins.getInstanceOrNull();
+            if (jenkins == null) {
+                return "";
+            }
+
+            PluginWrapper plugin = jenkins.pluginManager.getPlugin(STACKROX_CONTAINER_IMAGE_SCANNER);
+            if (plugin == null) {
+                return "";
+            }
+            return String.format("%s:%s", plugin.getVersion(), Jenkins.getVersion());
```

Comment:
I think `<plugin-version>:<jenkins-version>` should not allow spaces. Let's replace spaces in both `getVersion()` results here?

Reply:
Fixed